components/ratgdo/ratgdo.cpp

```diff
@@ -266,8 +266,8 @@ namespace ratgdo {
     {
         ESP_LOGD(TAG, "Schedule position sync: delta %f, start position: %f, start moving: %d",
             this->door_move_delta, this->door_start_position, this->door_start_moving);
-        auto duration = this->door_move_delta > 0 ? *this->opening_duration : -*this->closing_duration;
-        auto count = int(1000 * duration * this->door_move_delta / update_period);
+        auto duration = this->door_move_delta > 0 ? *this->opening_duration : *this->closing_duration;        
+        auto count = int(1000 * duration / update_period);
```

Comment:
I realized that it's better to schedule the door sync closure for the full open/close duration and have it cancelled when target is reached. Otherwise, if for example the door stop command fails, the position in HA stops at some percentage while the door fully closes or opens. This way the position in HA would be consistent with the actual door position.